WordPressAuthenticatorTests/Authenticator/WordPressAuthenticatorTests.swift

```diff
@@ -55,4 +65,190 @@ class WordPressAuthenticatorTests: XCTestCase {
+
+        authenticator.supportPushNotificationCleared()
+
+        waitForExpectations(timeout: timeInterval, handler: nil)
+    }
+
+//MARK: View Tests
+    func testWordpressAuthIsAuthenticationViewController() {
+        let loginViewcontroller = LoginViewController()
+        let nuxViewController = NUXViewController()
+        let nuxTableViewController = NUXTableViewController()
+        let basicViewController = UIViewController()
+
+
+        XCTAssertTrue(WordPressAuthenticator.isAuthenticationViewController(loginViewcontroller))
+        XCTAssertTrue(WordPressAuthenticator.isAuthenticationViewController(nuxViewController))
+        XCTAssertTrue(WordPressAuthenticator.isAuthenticationViewController(nuxTableViewController))
+        XCTAssertFalse(WordPressAuthenticator.isAuthenticationViewController(basicViewController))
+    }
+
+    func testShowLoginFromPresenterReturnsLoginInitialVC() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+
+        WordPressAuthenticator.showLoginFromPresenter(presenterSpy, animated: true)
+        wait(for: [expectation], timeout: 3)
+
+        XCTAssertTrue(presenterSpy.presentedVC is LoginNavigationController)
+    }
+
+    func testShowLoginForJustWPComPresentsCorrectVC() {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+
+        WordPressAuthenticator.showLoginForJustWPCom(from: presenterSpy)
+        wait(for: [expectation], timeout: 3)
+
+        XCTAssertTrue(presenterSpy.presentedVC is LoginNavigationController)
+    }
+
+    func testSignInForWPOrgReturnsVC() {
+        let vc = WordPressAuthenticator.signinForWPOrg()
+
+        XCTAssertTrue(vc is LoginSiteAddressViewController)
+    }
+
+    func testShowLoginForJustWPComSetsMetaProperties() throws {
+        let presenterSpy = ModalViewControllerPresentingSpy()
+        let expectation = XCTNSPredicateExpectation(predicate: NSPredicate(block: { (_, _) -> Bool in
+            return presenterSpy.presentedVC != nil
+        }), object: .none)
+
+        WordPressAuthenticator.showLoginForJustWPCom(from: presenterSpy,
+                                                     xmlrpc: "https://example.com/xmlrpc.php",
+                                                     username: "username",
+                                                     connectedEmail: "email-address@example.com")
+
+        let navController = try XCTUnwrap(presenterSpy.presentedVC as? LoginNavigationController)
+        let controller = try XCTUnwrap(navController.viewControllers.first as? LoginEmailViewController)
+
+        wait(for: [expectation], timeout: 3)
```

Comment:
Should this use `timeInterval` (or `timeout` as per my earlier suggestion), too?